controllers/application_controller.go

```diff
@@ -73,6 +77,44 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return reconcile.Result{}, err
 	}
 
+	// Check if the Application CR is under deletion
+	// If so: Remove the GitOps repo (if generated) and remove the finalizer.
+	if application.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !containsString(application.GetFinalizers(), appFinalizerName) {
+			if err := r.AddFinalizer(ctx, &application); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if containsString(application.GetFinalizers(), appFinalizerName) {
+			// A finalizer is present for the Application CR, so make sure we do the necessary cleanup steps
+			if err := r.Finalize(&application); err != nil {
+				finalizeCount, err := getFinalizeCount(&application)
+				if err == nil && finalizeCount < 5 {
+					// The Finalize function failed, so increment the finalize count and return
+					setFinalizeCount(&application, finalizeCount+1)
+					err := r.Update(ctx, &application)
+					if err != nil {
+						log.Error(err, "Error incrementing finalizer count on resource")
+					}
+					return ctrl.Result{}, nil
```

Comment:
Nope, not needed here. Since the resource is being updated, the reconcile loop will still be called again.  `Requeue: true` should only be needed if we need to restart the reconcile loop for some other reasons (i.e. to refresh the cached Kube client).